src/Compiler/AbstractIL/ilwritepdb.fs

```diff
@@ -961,98 +1201,135 @@ let logDebugInfo (outfile: string) (info: PdbData) =
-      fprintfn sw "     Params: %A" [ for p in meth.Params -> sprintf "%d: %s" p.Index p.Name ]
-      fprintfn sw "     Range: %A" (meth.DebugRange |> Option.map (fun (f, t) ->
-                                      sprintf "[%d,%d:%d] - [%d,%d:%d]" f.Document f.Line f.Column t.Document t.Line t.Column))
-      fprintfn sw "     Points:"
-
-      for sp in meth.DebugPoints do
-        fprintfn sw "      - Doc: %d Offset:%d [%d:%d]-[%d-%d]" sp.Document sp.Offset sp.Line sp.Column sp.EndLine sp.EndColumn
 
-      // Walk through the tree of scopes and write all variables
-      fprintfn sw "     Scopes:"
-      let rec writeScope offs (scope: PdbMethodScope) =
-        fprintfn sw "      %s- [%d-%d]" offs scope.StartOffset scope.EndOffset
-        if scope.Locals.Length > 0 then
-          fprintfn sw "      %s  Locals: %A" offs [ for p in scope.Locals -> sprintf "%d: %s" p.Index p.Name ]
-
-        for child in scope.Children do writeScope (offs + "  ") child
+    for meth in info.Methods do
+        fprintfn sw " %s" meth.MethName
+        fprintfn sw "     Params: %A" [ for p in meth.Params -> sprintf "%d: %s" p.Index p.Name ]
+
+        fprintfn
+            sw
+            "     Range: %A"
+            (meth.DebugRange
+             |> Option.map (fun (f, t) ->
+                 sprintf "[%d,%d:%d] - [%d,%d:%d]" f.Document f.Line f.Column t.Document t.Line t.Column))
+
+        fprintfn sw "     Points:"
+
+        for sp in meth.DebugPoints do
+            fprintfn
+                sw
+                "      - Doc: %d Offset:%d [%d:%d]-[%d-%d]"
+                sp.Document
+                sp.Offset
+                sp.Line
+                sp.Column
+                sp.EndLine
+                sp.EndColumn
+
+        // Walk through the tree of scopes and write all variables
+        fprintfn sw "     Scopes:"
+
+        let rec writeScope offs (scope: PdbMethodScope) =
+            fprintfn sw "      %s- [%d-%d]" offs scope.StartOffset scope.EndOffset
+
+            if scope.Locals.Length > 0 then
+                fprintfn sw "      %s  Locals: %A" offs [ for p in scope.Locals -> sprintf "%d: %s" p.Index p.Name ]
+
+            for child in scope.Children do
+                writeScope (offs + "  ") child
+
+        match meth.RootScope with
+        | None -> ()
+        | Some rootscope -> writeScope "" rootscope
 
-      match meth.RootScope with
-      | None -> ()
-      | Some rootscope -> writeScope "" rootscope
-      fprintfn sw ""
+        fprintfn sw ""
 
 let rec allNamesOfScope acc (scope: PdbMethodScope) =
     let acc = (acc, scope.Locals) ||> Array.fold (fun z l -> Set.add l.Name z)
     let acc = (acc, scope.Children) ||> allNamesOfScopes
     acc
+
 and allNamesOfScopes acc (scopes: PdbMethodScope[]) =
     (acc, scopes) ||> Array.fold allNamesOfScope
 
 let rec pushShadowedLocals (stackGuard: StackGuard) (localsToPush: PdbLocalVar[]) (scope: PdbMethodScope) =
-    stackGuard.Guard <| fun () ->
```

Comment:
The chagnes here are a bit painful, creating an indentation that wasn't here before.  Might be better to have a separate method containing the stack-guard